lib/Dialect/FIRRTL/Transforms/LowerXMR.cpp

```diff
@@ -199,6 +197,14 @@ class LowerXMRPass : public LowerXMRBase<LowerXMRPass> {
 
   // Replace the RefResolveOp with verbatim op representing the XMR.
   LogicalResult handleRefResolve(RefResolveOp resolve) {
+    if (!resolve.getType().getBitWidthOrSentinel()) {
+      // Donot emit 0 width XMRs, instead rely on `LowerToHW` to remove these
+      // temporary 0-width wires.
+      ImplicitLocOpBuilder builder(resolve.getLoc(), resolve);
+      auto dummyWire = builder.create<WireOp>(resolve.getType());
```

Comment:
What about a zero-width constant zero? instead of a zero-width wire?

I'm somewhat concerned that this zero-width wire is undriven.

Reply:
What happens if this is a vector, `firrtl.ref<vector<uint<0>,2>`

Reply:
🤔  My main concern is that this is dropping an undriven wire and you're indicating that we don't have the ability to express a vector literal. Perhaps we continue with the wire and drive it with constant zeros (doing the annoying expansion to do this).

Reply:
Usually I have used `bitcast(constant zero)` to create aggregate constant. Probably we can use that here?

Reply:
Nice, thanks @uenoku , that's a nice idea. Will do that.